### PR TITLE
Send MapServer substitution only on MapServer with standard auth

### PR DESCRIPTION
**Pull request by sbrunner**
#8400 also for the GetCapabilities

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/__init__.py [hunk 1]
```diff
@@ -234,7 +234,9 @@ def get_user_from_request(request, username=None):
             if username is not None:
                 # We know we will need the role object of the
                 # user so we use joined loading
-                request.user_ = DBSession.query(User).filter_by(username=username).first()
+                request.user_ = (
+                    DBSession.query(User).filter_by(username=username).options(joinedload("roles")).first()
```
Comment (ochriste) [inline geoportal/c2cgeoportal_geoportal/__init__.py:238]: why did you change the relation loading strategy? for performance ?

Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/__init__.py:238]: It's good for performance, but I do that because without that, I get this kind of error
https://github.com/camptocamp/c2cgeoportal/runs/2945488684?check_suite_focus=true
```
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/views/theme.py:1100: in themes
    return get_theme()
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/views/theme.py:1038: in get_theme
    l.name for l in get_protected_layers(self.request, [ogc_server.id]).values()
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/lib/layers.py:74: in get_protected_layers
    q = get_protected_layers_query(request, ogc_server_ids, what=main.LayerWMS)
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/lib/layers.py:52: in get_protected_layers_query
    q = _get_layers_query(request, what)
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/lib/layers.py:44: in _get_layers_query
    q = q.filter(main.Role.id.in_(get_roles_id(request)))
/opt/c2cgeoportal/geoportal/c2cgeoportal_geoportal/lib/__init__.py:277: in get_roles_id
    result.extend([r.id for r in request.user.roles])
/usr/local/lib/python3.7/dist-packages/sqlalchemy/orm/attributes.py:286: in __get__
    return self.impl.get(instance_state(instance), dict_)
/usr/local/lib/python3.7/dist-packages/sqlalchemy/orm/attributes.py:722: in get
    value = self.callable_(state, passive)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
            raise orm_exc.DetachedInstanceError(
                "Parent instance %s is not bound to a Session; "
                "lazy load operation of attribute '%s' cannot proceed"
>               % (orm_util.state_str(state), self.key)
            )
E           sqlalchemy.orm.exc.DetachedInstanceError: Parent instance <User at 0x7f24aff93950> is not bound to a Session; lazy load operation of attribute 'roles' cannot proceed (Background on this error at: http://sqlalche.me/e/bhk3)
```

**Review by arnaud-morvan — APPROVED** [review]
